arex-instrumentation/dynamic/arex-cache/src/main/java/io/arex/inst/cache/util/CacheLoaderUtil.java

```diff
@@ -59,30 +56,38 @@ private static String generateNameWithNoReference(int loaderHashCode, Class<?> l
         return loaderClassName;
     }
 
-    private static boolean isNotAbstractOrInterface(Class<?> clazz) {
-        if (clazz == null) {
-            return true;
-        }
-        int modifiers = clazz.getModifiers();
-        return !Modifier.isAbstract(modifiers) && !clazz.isInterface();
-    }
-
     private static Field getReferenceField(Class<?> loaderClass) {
         try {
-            Field referenceField = loaderClass.getDeclaredField("this$0");
-            referenceField.setAccessible(true);
+            Field referenceField = null;
+            for (Field field : loaderClass.getDeclaredFields()) {
+                if (StringUtil.startWith(field.getName(), EXTERNAL_VARIABLE_REFERENCE_IDENTIFIER)) {
+                    referenceField = field;
+                    referenceField.setAccessible(true);
+                    return referenceField;
+                }
+                if (StringUtil.equals(field.getName(), EXTERNAL_INSTANCE_REFERENCE_IDENTIFIER)) {
+                    referenceField = field;
+                    referenceField.setAccessible(true);
```

Comment:
missing `return` here？

Reply:
the external traversal reference field has the highest priority. If it is recognized, it can be returned. If not, all fields will be traversed.